src/Chr.Avro.Binary/BinaryDeserializerBuilder.cs

```diff
@@ -1570,6 +1565,44 @@ public override Delegate BuildDelegate(TypeResolution resolution, Schema schema,
 
             return compiled;
         }
+               
+        /// <summary>
+        /// Whether the resolved Constructor matches a RrecordSchemas fields.
+        /// </summary>
+        /// <param name="constructor">
+        /// The ConstructorResolution to check for match.
+        /// </param>
+        /// <param name="recordFields">
+        /// The RecordSchemas fields to match the against constructors parameters. 
+        /// </param>
+        /// <returns></returns>
+        public bool IsMatch(ConstructorResolution constructor, IList<RecordField> recordFields)
```

Comment:
public -> protected, also would do `ICollection` unless there’s precedent for `IList` somewhere else

Reply:
I was originally doing this with a for loop instead of a foreach loop and needed to be able to access the indexer. Changed to ICollection since that isn't how I ended up implementing it in the end.